src/GraphQL.SystemTextJson/GraphQLSerializer.cs

```diff
@@ -78,9 +78,14 @@ public GraphQLSerializer(Action<JsonSerializerOptions> configureSerializerOption
         /// <param name="serializerOptions">Specifies the JSON serializer settings</param>
         public GraphQLSerializer(JsonSerializerOptions serializerOptions)
         {
+#if NET5_0_OR_GREATER
+            // clone serializerOptions
+            SerializerOptions = new(serializerOptions ?? throw new ArgumentNullException(nameof(serializerOptions)));
```

Comment:
Hmm, new API since NET5?

Reply:
It worth to add some notes like this in summary or for `serializeOptions` argument:
> Note that `JsonSerializerOptions` is designed in such a way to reuse created objects. This leads to a massive speedup in subsequent calls to the serializer. The downside is you can’t change properties on the options object after you’ve passed it in a Serialize()/Deserialize() call. You’ll get the exception: System.InvalidOperationException: Serializer options cannot be changed once serialization or deserialization has occurred. To get around this problem we create a copy of options on NET5 platform and above. Passed options object remains unchanged and all possible changes you made are unobserved.

Reply:
What about TFMs < 5 ?

Reply:
Well, I'm just fixing what I can easily.  Fixing the bug for older version is up for grabs.

Reply:
Sure we can add the comment in there somewhere.  Not really sure people care about 'what makes it tick', so long as it does "tick", but sure.

Reply:
I suggest a slight change: "and all possible changes you made" to "and any changes you make"

Reply:
Updated -- added to constructors that accept `JsonSerializerOptions` (not `Action<JsonSerializerOptions>`)

Reply:
> Not really sure people care about 'what makes it tick', so long as it does "tick", but sure.

This comment is needed for those cases if someone thought to change the values of options later. This is a warning. Not everyone knows how STJ works undercover.